### PR TITLE
fix(ct): unsuccessful login error message

### PR DESCRIPTION
**Pull request by dawid-ziobro**
## Description

graphQL erro was not cough before returning data. Fix is to check it first and throw error if data is wrong.

## Related Issue

closes OS-170

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

#### Changelog
- [x] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.

#### Tests
- [ ] I have written test cases for my code
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

#### Code standards
- [x] My code follows the code style of this project.

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

#### Docs
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/commercetools/composables/src/useUser/authenticate.ts [hunk 1]
```diff
@@ -3,10 +3,19 @@ import { CustomerSignMeInDraft, CustomerSignMeUpDraft } from '../types/GraphQL';
 
 type UserData = CustomerSignMeUpDraft | CustomerSignMeInDraft;
 
+const validateUserResponse = (response) => {
+  if (response?.data?.user) {
+    return response.data.user;
+  }
+
```
Comment (RafalDobrowolski) [inline packages/commercetools/composables/src/useUser/authenticate.ts:10]: remove empty line what do you think ? :)

File: packages/commercetools/composables/src/useUser/authenticate.ts [hunk 2]
```diff
@@ -3,10 +3,19 @@ import { CustomerSignMeInDraft, CustomerSignMeUpDraft } from '../types/GraphQL';
...
+  throw response;
+};
+
 export const authenticate = async (userData: UserData, fn) => {
   try {
     const userResponse = await fn(userData);
-    return userResponse.data.user;
+
```
Comment (RafalDobrowolski) [inline packages/commercetools/composables/src/useUser/authenticate.ts:17]: same as above

File: packages/commercetools/composables/src/useUser/authenticate.ts (hunk 1 above)
Comment (dawid-ziobro) [reply packages/commercetools/composables/src/useUser/authenticate.ts:10]: It's added on purpose to increase readability

File: packages/commercetools/composables/src/useUser/authenticate.ts (hunk 2 above)
Comment (dawid-ziobro) [reply packages/commercetools/composables/src/useUser/authenticate.ts:17]: same as above :D

**Review by RafalDobrowolski — APPROVED** [review]

**Review by lukaszjedrasik — APPROVED** [review]

**Review by igorwojciechowski — APPROVED** [review]

**Pull request merged**